sources/Core/Email/EmailLaminas.php

```diff
@@ -293,24 +293,59 @@ protected function EmbedInlineImages(string &$sBody)
 		return $aImagesParts;
 	}
 
-	public function Send(&$aIssues, $bForceSynchronous = false, $oLog = null)
+	/**
+	 * Sends an e-mail.
+	 *
+	 * @param String[] $aIssues Array to add any potentially encountered issues to.
+	 * @param Integer $iSyncAsync Specify whether the e-mail will be sent per default configuration, or whether there will be a forced (a)synchronous sending. One of Email::ENUM_SEND_* const
+	 * @param Object|null $oLog Log
+	 *
+	 * @details By default, the send method will respect the preference to send e-mails in an (a)synchronous way as defined in the iTop configuration by the administrator.
+	 * In some use cases, it may be necessary to override this behavior. For example, for some tests it may be best if e-mails are always sent instantly (synchronous).
+	 * Asynchronous may be preferred when sending a lot of bulk e-mails at once, to avoid hitting rate limits of e-mail providers (e.g. customer survey extension).
+	 *
+	 * @return
+	 */
+	public function Send(&$aIssues, $iSyncAsync = Email::ENUM_SEND_DEFAULT, $oLog = null)
 	{
 		//select a default sender if none is provided.
 		if (empty($this->m_aData['from']['address']) && !empty($this->m_aData['to'])) {
 			$this->SetRecipientFrom($this->m_aData['to']);
 		}
 
-		if ($bForceSynchronous) {
+		// In previous iTop versions, $iSyncAsync was $bForceSynchronous. To retain backward compatibility, this check is in place.
+		// @todo Decide: log a deprecation warning or something? Or turn $iSyncAsync officially into a mixed value?
+		if(is_bool($iSyncAsync) == true && $iSyncAsync == true) {
```

Comment:
```suggestion
		if(is_bool($iSyncAsync) && $iSyncAsync == true) {
```
Or maybe just like this (less explicit):
```suggestion
		if($iSyncAsync === true) {
```

Reply:
I would be in favor of the second proposal :)

Reply:
Closing conversation as the code as evolved since then